rclpy/actions/minimal_action_server/examples_rclpy_minimal_action_server/server.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+
+from example_interfaces.action import Fibonacci
+
+import rclpy
+from rclpy.action import ActionServer, CancelResponse, GoalResponse
+from rclpy.callback_groups import ReentrantCallbackGroup
+from rclpy.executors import MultiThreadedExecutor
+from rclpy.node import Node
+
+
+class MinimalActionServer(Node):
+
+    def __init__(self):
+        super().__init__('minimal_action_server')
+
+        self._action_server = ActionServer(
+            self,
+            Fibonacci,
+            'fibonacci',
+            execute_callback=self.execute_callback,
+            callback_group=ReentrantCallbackGroup(),
+            goal_callback=self.goal_callback,
+            cancel_callback=self.cancel_callback)
+
+    def destroy(self):
+        self._action_server.destroy()
+        super().destroy_node()
+
+    def goal_callback(self, goal_request):
+        """Accepts or rejects a client request to begin an action."""
+        # This server allows multiple goals in parallel
+        self.get_logger().info('Received goal request')
+        return GoalResponse.ACCEPT_AND_EXECUTE
+
+    def cancel_callback(self, goal_handle):
+        """Accepts or rejects a client request to cancel an action."""
+        self.get_logger().info('Received cancel request')
```

Comment:
When running the cancel example this is the last log message I see. Both the client and the server continue to run. I was expecting to see the server say `Goal canceld` from line 67.

```
$ ros2 run examples_rclpy_minimal_action_client client_cancel 
[INFO] [minimal_action_client]: Waiting for action server...
[INFO] [minimal_action_client]: Sending goal request...
[INFO] [minimal_action_client]: Goal accepted :)
[INFO] [minimal_action_client]: Received feedback: [0, 1, 1]
[INFO] [minimal_action_client]: Received feedback: [0, 1, 1, 2]
[INFO] [minimal_action_client]: Received feedback: [0, 1, 1, 2, 3]
[INFO] [minimal_action_client]: Canceling goal
```

```
$ ros2 run examples_rclpy_minimal_action_server server
[INFO] [minimal_action_server]: Received goal request
[INFO] [minimal_action_server]: Executing goal...
[INFO] [minimal_action_server]: Publishing feedback: [0, 1, 1]
[INFO] [minimal_action_server]: Publishing feedback: [0, 1, 1, 2]
[INFO] [minimal_action_server]: Publishing feedback: [0, 1, 1, 2, 3]
[INFO] [minimal_action_server]: Received cancel request
```

Reply:
Fixed in ce7bb67

I was making a synchronous call in a callback (which is no good, [as documented](https://github.com/ros2/rclpy/blob/4028bc6ab204cea5d44d8f1c14faa5fd4f26562d/rclpy/rclpy/action/client.py#L75) :disappointed:)